From 1dab7bf2645f0da38c73af270a7991a40b81672c Mon Sep 17 00:00:00 2001
From: Corentin Kerisit <corentin.kerisit@gmail.com>
Date: Tue, 13 May 2025 17:39:39 +0200
Subject: [PATCH 3/3] Adding libc-modules.h

This file is only used while compiling the small part of the libc
we build from source.

It checks, which libc module (binary or library) is being built
to define defines and constants.

The current version is from 2.41 but if any compilation error
happens because some old glibc code is referring to libc modules
that do not exist anymore, it should be fine to just define it
again to a random number.

This is because the only libc module we compile is the libc.
---
 include/libc-modules.h | 40 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 include/libc-modules.h

diff --git a/include/libc-modules.h b/include/libc-modules.h
new file mode 100644
index 0000000000..c3400abbdc
--- /dev/null
+++ b/include/libc-modules.h
@@ -0,0 +1,40 @@
+/* AUTOGENERATED BY gen-libc-modules.awk, DO NOT EDIT.  */
+
+#define MODULE_iconvprogs 1
+#define MODULE_iconvdata 2
+#define MODULE_ldconfig 3
+#define MODULE_libmemusage 4
+#define MODULE_libpcprofile 5
+#define MODULE_librpcsvc 6
+#define MODULE_locale_programs 7
+#define MODULE_memusagestat 8
+#define MODULE_nonlib 9
+#define MODULE_nscd 10
+#define MODULE_extramodules 11
+#define MODULE_libnldbl 12
+#define MODULE_libsupport 13
+#define MODULE_testsuite 14
+#define MODULE_testsuite_internal 15
+#define MODULE_LIBS_BEGIN 16
+#define MODULE_rtld 17
+#define MODULE_libc 18
+#define MODULE_libBrokenLocale 19
+#define MODULE_libpthread 20
+#define MODULE_libthread_db 21
+#define MODULE_libcrypt 22
+#define MODULE_libdl 23
+#define MODULE_libgcc_s 24
+#define MODULE_libnsl 25
+#define MODULE_libc_malloc_debug 26
+#define MODULE_libutil 27
+#define MODULE_libnss_ldap 28
+#define MODULE_libnss_dns 29
+#define MODULE_libnss_compat 30
+#define MODULE_libmvec 31
+#define MODULE_libresolv 32
+#define MODULE_libnss_db 33
+#define MODULE_libm 34
+#define MODULE_libnss_files 35
+#define MODULE_librt 36
+#define MODULE_libnss_hesiod 37
+#define MODULE_libanl 38
-- 
2.39.3 (Apple Git-146)

